cmd/billing/billing.go

```diff
@@ -0,0 +1,197 @@
+// licenses this file to you under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package billing
+
+import (
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	ybmAuthClient "github.com/yugabyte/ybm-cli/internal/client"
+	ybmclient "github.com/yugabyte/yugabytedb-managed-go-client-internal"
+)
+
+var BillingCmd = &cobra.Command{
+	Use:   "billing",
+	Short: "Manage billing operations for the account in YugabyteDB Managed",
+	Long:  "Manage billing operations for the account in YugabyteDB Managed",
+	Run: func(cmd *cobra.Command, args []string) {
+		cmd.Help()
+	},
+}
+
+var listUsageCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List billing usage data available for the account in YugabyteDB Managed",
+	Long:  "List billing usage data available for the account in YugabyteDB Managed",
+	Run: func(cmd *cobra.Command, args []string) {
+		authApi, err := ybmAuthClient.NewAuthApiClient()
+		if err != nil {
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+		authApi.GetInfo("", "")
+		startDate, _ := cmd.Flags().GetString("start")
+		endDate, _ := cmd.Flags().GetString("end")
+		clusters, _ := cmd.Flags().GetStringSlice("clusters")
+		format, _ := cmd.Flags().GetString("format")
+
+		respC, r, err := authApi.ListClustersByDateRange(startDate, endDate).Execute()
+		if err != nil {
+			logrus.Debugf("Full HTTP response: %v", r)
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+		clusterData := respC.Data.GetClusters()
+		var clusterUUIDMap = make(map[string]string)
+
+		for _, cluster := range clusterData {
+			clusterName := cluster.Name
+			clusterUUIDMap[clusterName] = cluster.Id
+		}
+
+		var selectedUUIDs []string
+		if len(clusters) > 0 {
+			selectedUUIDs = make([]string, 0, len(clusters))
+			for _, clusterName := range clusters {
+				if clusterID, ok := clusterUUIDMap[clusterName]; ok {
+					selectedUUIDs = append(selectedUUIDs, clusterID)
+				} else {
+					logrus.Warnf("Cluster name '%s' not found in clusterUUIDMap", clusterName)
+				}
+			}
+		} else {
+			selectedUUIDs = nil
+		}
+
+		resp, r, err := authApi.GetBillingUsage(startDate, endDate, selectedUUIDs).Execute()
+
+		if err != nil {
+			logrus.Debugf("Full HTTP response: %v", r)
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+		usageData := resp.GetData()
+
+		switch strings.ToLower(format) {
+		case "csv":
+			if err := outputCSV(usageData, "usage.csv"); err != nil {
```

Comment:
Can we have an option for the filename ? without the extension of course

Reply:
Done -> added output flag for filename.